editor/components/post-permalink/editor.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies
+ */
+import { withDispatch } from '@wordpress/data';
+import { Component, compose } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+
+/**
+ * Internal Dependencies
+ */
+import './style.scss';
+import { getEditedPostAttribute } from '../../store/selectors';
+
+class PostPermalinkEditor extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			editedPostName: '',
+		};
+
+		this.onSavePermalink = this.onSavePermalink.bind( this );
+	}
+
+	/**
+	 * Returns the permalink parts to populate the form.
+	 *
+	 * @return {Object} The prefix, suffix, and postName for the form.
+	 */
+	getPermalinkParts() {
+		const [ template, postName ] = this.props.samplePermalinkData;
+		const [ samplePermalinkPrefix, samplePermalinkSuffix ] = template.split( /%(?:postname|pagename)%/ );
+
+		return {
+			samplePermalinkPrefix,
+			samplePermalinkSuffix,
+			postName,
+		};
+	}
+
+	onSavePermalink() {
+		const postName = this.state.editedPostName.replace( /\s+/g, '-' );
+		const [ template, oldPostName ] = this.props.samplePermalinkData;
+
+		this.props.onSave();
+
+		if ( ! postName || postName === oldPostName ) {
+			return;
+		}
+
+		this.props.editPost( {
+			slug: postName,
+			sample_permalink: [ template, postName ],
```

Comment:
Something feels a bit off about the fact that we update the post's `sample_permalink`, but I can't quite put my finger on it. It's likely that it feels duplicated with `slug`, and that `template` is not really an editable field.

Reply:
`sample_permalink` isn't editable in the API, but it's useful to be able to update the local data model with the correct value, until the post is saved.